datadog/dogshell/wrap.py

```diff
@@ -219,25 +221,44 @@ def build_event_body(cmd, returncode, stdout, stderr, notifications):
         )
 
 
+def generate_warning_codes(option, opt, options_warning):
+    try:
+        # options_warning is a string e.g.: --warning_codes 123,456,789
+        # we need to create a list from it
+        warning_codes = options_warning.split(",")
+        return warning_codes
+    except ValueError:
+        raise optparse.OptionValueError("option %s: invalid warning codes value(s): %r" % (opt, options_warning))
+
+
+class WarningOption(optparse.Option):
```

Comment:
The type name might not be that good since it's going to be your option type with all your types.
`DogshellOption` might be better?

Reply:
Since it's for `dogwrap` only, we may even want to call this `DogwrapOption`.